history_test.go

```diff
@@ -16,28 +17,57 @@ type testWriteCloser struct {
 func (wrc *testWriteCloser) Close() error { return nil }
 
 func TestExecute(t *testing.T) {
-	command := "echo testing"
-	var output bytes.Buffer
-	r, err := history.NewRecorder()
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = r.EnsureHistoryFileOpen()
-	if err != nil {
-		fmt.Fprint(r.Stdout, err)
-	}
-	r.Stdout = &output
-	err = r.Execute(command)
-	if err != nil {
-		t.Fatal(err)
+	testCases := []struct {
+		command     string
+		desc        string
+		errExpected bool
+		output      bytes.Buffer
+		want        string
+	}{
+		{
+			desc:    "Simple Echo",
+			command: "echo testing",
+			want:    "testing\n",
+		},
+		{
+			desc:        "Non-existing command",
+			command:     "abc",
+			want:        "exec: \"abc\": executable file not found in $PATH",
+			errExpected: true,
+		},
 	}
-	wantOutput := "testing\n"
-	if !cmp.Equal(wantOutput, output.String()) {
-		t.Error(cmp.Diff(wantOutput, output.String()))
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			r, err := history.NewRecorder()
```

Comment:
Do we need a new Recorder for every test case, or could we just execute all the test commands on a single Recorder?

Reply:
One should be good, thx.